services/transactionpool/test/add.go

```diff
@@ -46,4 +46,19 @@ var _ = Describe("transaction pool", func() {
 
 	})
 
+	It("dont forward an invalid transaction with gossip", func() {
+
+		tx := test.TransferTransaction().WithInvalidContent().Build()
+
+		gossip.When("BroadcastForwardedTransactions", mock.Any).Return(&gossiptopics.EmptyOutput{}, nil).Times(0)
+
+		_, err := service.AddNewTransaction(&services.AddNewTransactionInput{
+			SignedTransaction: tx,
+		})
+
+		Expect(err).To(HaveOccurred())
+		Expect(gossip).To(test.ExecuteAsPlanned())
+
+	})
+
```

Comment:
For the sake of completeness, I'd also add a happy path test here. Seems a bit lonely now :-(

Reply:
This is the first test...